src/twisted/application/runner/_runner.py

```diff
@@ -31,19 +29,73 @@ class Runner(object):
     log = Logger()
 
 
-    def __init__(self, options):
+    def __init__(
+        self,
+        reactor=None,
+        pidFile=nonePIDFile, kill=False,
+        defaultLogLevel=LogLevel.info,
+        logFile=stderr, fileLogObserverFactory=textFileLogObserver,
+        whenRunning=lambda **_: None, whenRunningArguments={},
+        reactorExited=lambda **_: None, reactorExitedArguments={},
+    ):
         """
-        @param options: Configuration options for this runner.
-        @type options: mapping of L{RunnerOptions} to values
+        @param reactor: The reactor to start and run the application in.
+        @type reactor: L{IReactorCore}
+
+        @param pidFile: The file to store the running process ID in.
+        @type pidFile: L{IPIDFile}
+
+        @param kill: Whether this runner should kill an existing running
+            instance of the application.
+        @type kill: L{bool}
+
+        @param defaultLogLevel: The default log level to start the logging
+            system with.
+        @type defaultLogLevel: L{constantly.NamedConstant} from L{LogLevel}
+
+        @param logFile: A file stream to write logging output to.
+        @type logFile: writable file-like object
+
+        @param fileLogObserverFactory: A factory for the file log observer to
+            use when starting the logging system.
+        @type pidFile: callable that takes a single writable file-like object
+            argument and returns a L{twisted.logger.FileLogObserver}
+
+        @param whenRunning: Hook to call after the reactor is running;
+            this is where the application code that relies on the reactor gets
+            called.
+        @type whenRunning: callable that takes the keyword arguments specified
+            by C{whenRunningArguments}
+
+        @param whenRunningArguments: Keyword arguments to pass to
+            C{whenRunning} when it is called.
+        @type whenRunningArguments: L{dict}
+
+        @param reactorExited: Hook to call after the reactor exits.
+        @type reactorExited: callable that takes the keyword arguments
+            specified by C{reactorExitedArguments}
+
+        @param reactorExitedArguments: Keyword arguments to pass to
+            C{reactorExited} when it is called.
+        @type reactorExitedArguments: L{dict}
         """
-        self.options = options
+        self.reactor                = reactor
```

Comment:
More germane to the actual change here - do we want all these attributes to be public?  Some of them possibly should be (`reactor`?), but, clearly some are private.